sympy/matrices/eigen.py

```diff
@@ -150,6 +161,36 @@ def _eigenvals(M, error_when_incomplete=True, **flags):
         return [simplify(value) for value in eigs]
 
 
+def _eigenspace(M, eigenval, iszerofunc=_iszero, simplify=False):
+    """Get a basis for the eigenspace for a particular eigenvalue
+
+    The nullspace for a real eigenvalue should be non-trivial.
```

Comment:
I would suggest moving this into the code since it is an "implementation detail" (as I understand it): you try without simplificaiton since that might be a lot quicker and if you succeed, great; and if not you use the simplification option.